### PR TITLE
log-backup: added more metrics and hint; fixed a bug may cause inf loop

### PR DESCRIPTION
**Pull request by YuJuncen**
Signed-off-by: Yu Juncen <yujuncen@pingcap.com>

### What problem does this PR solve?

Issue Number: ref #29501

Problem Summary:
Before, we may get stuck after some region get merged because the poor exit reason of `RegionIter`.

### What is changed and how it works?
Now, region iterator would exit if `infScanFinished` is `true` no matter the end key is inf.

### Check List

Tests 

- [x] Unit test (TBD)
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/streamhelper/advancer.go [hunk 1]
```diff
@@ -230,6 +242,14 @@ func (c *CheckpointAdvancer) CalculateGlobalCheckpointLight(ctx context.Context)
 	if len(rsts) == 0 {
 		return 0, nil
 	}
+	samples := rsts
+	if len(rsts) > 3 {
+		samples = rsts[:3]
+	}
+	for _, sample := range samples {
+		log.Info("sample range.", zap.Object("range", sample.Zap()), zap.Int("total-len", len(rsts)))
```
[the remark this answers is not in the document]
Comment (YuJuncen) [reply br/pkg/streamhelper/advancer.go:250]: It would only print 3 logs each tick.

[the remark this answers is not in the document]
Comment (3pointer) [reply br/pkg/streamhelper/advancer.go:239]: Then it's better to give a clear msg like "[log-backup] advance calculate sample ranges in one tick"

**Comment by 3pointer** [comment]
rest LGTM, I think we can give a meaningful log messages. because not only we will check the log. every one can see the log too. some log is not easy for other people to understand.

**Review by Leavrth — APPROVED** [review]

File: metrics/metrics.go [hunk 2]
```diff
@@ -195,6 +195,8 @@ func RegisterMetrics() {
 	prometheus.MustRegister(AdvancerOwner)
 	prometheus.MustRegister(AdvancerTickDuration)
 	prometheus.MustRegister(GetCheckpointBatchSize)
+	prometheus.MustRegister(RegionCheckpointRequest)
```
Comment (3pointer) [inline metrics/metrics.go:198]: shall we add metrics json file in this PR?

Comment (YuJuncen) [reply metrics/metrics.go:198]: Those metrics are added into the TiKV panel...

Comment (3pointer) [reply metrics/metrics.go:198]: ok

**Review by 3pointer — APPROVED** [review]
LGTM

**Pull request merged**